src/Frontend-GNOME/Views/MessageTextView.cs

```diff
@@ -406,6 +407,25 @@ public void AddMessage(MessageModel msg, bool addLinebreak, bool showTimestamps)
                     } else {
                         buffer.Insert(ref iter, fmsgti.Text);
                     }
+                } else if (msgPart is ImageMessagePartModel) {
+                    var imgpart = (ImageMessagePartModel) msgPart;
+                    try {
+                        var uri = new Uri(imgpart.ImageFileName);
+                        if (uri.Scheme == "smuxi-emoji") {
+                            var emojiPath = System.IO.Path.Combine("/tmp/emoji/assets/images", uri.Host + ".png");
```

Comment:
Yeah, that was so I could see if it worked. This will change to make use of the icon cache.